src/Parser/Shaman/Restoration/Modules/Talents/EarthenWallTotem.js

```diff
@@ -12,79 +12,84 @@ import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
 
 import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
 
-import CooldownThroughputTracker from '../Features/CooldownThroughputTracker';
-
 class EarthenWallTotem extends Analyzer {
   static dependencies = {
-    cooldownThroughputTracker: CooldownThroughputTracker,
     abilityTracker: AbilityTracker,
     combatants: Combatants,
   };
 
-  activeEST = null;
+  prePullCast = true;
+  prePullCastHealth = 0; // Used to be boolean, but i need the number in case the player is Mag'har Orc
   potentialHealing = 0;
   healing = 0;
 
-
   on_initialized() {
-    this.active = this.combatants.selected.hasTalent(SPELLS.EARTHEN_SHIELD_TOTEM_TALENT.id);
+    this.active = this.combatants.selected.hasTalent(SPELLS.EARTHEN_WALL_TOTEM_TALENT.id);
   }
 
-  on_damage(event) {
+  on_toPlayerPet_damage(event) {
     const spellId = event.ability.guid;
 
-    if (!(spellId === SPELLS.EARTHEN_SHIELD_TOTEM_SELF_DAMAGE.id)) {
+    if (spellId !== SPELLS.EARTHEN_SHIELD_TOTEM_SELF_DAMAGE.id) {
       return;
     }
 
-    if (event.targetID === this.activeEST) {
-      this.healing += (event.amount || 0) + (event.overheal || 0) + (event.absorbed || 0);
+    if (this.prePullCast) {
+      this.potentialHealing += event.maxHitPoints; // this is taking the totems max HP, which is the same result as the players unless Mag'har Orc
+      this.prePullCast = false;
+      this.prePullCastHealth = event.maxHitPoints;
     }
 
+    this.healing += (event.amount || 0) + (event.overheal || 0) + (event.absorbed || 0);
```

Comment:
Damage events shouldn't have an overheal prop, and I don't think totems can get absorb shields (and if they can I don't think you'd want to count it as the totems healing anyway).

Reply:
yeah, when its a module that someone else wrote I often keep the oddities in case it does something I don't think about. I'll remove it in a minute so be ready for another travis error :P